pkg/stanza/fileconsumer/matcher/internal/filter/sort_test.go

```diff
@@ -240,3 +240,100 @@ func TestMTimeFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestExcludeOlderThanFilter(t *testing.T) {
+	twoHoursAgo := time.Now().Add(-2 * time.Hour)
+	threeHoursAgo := twoHoursAgo.Add(-1 * time.Hour)
+
+	cases := map[string]struct {
+		files            []string
+		fileMTimes       []time.Time
+		excludeOlderThan time.Duration
+
+		expect      []string
+		expectedErr string
+	}{
+		"no_files": {
+			files:            []string{},
+			fileMTimes:       []time.Time{},
+			excludeOlderThan: 2 * time.Hour,
+
+			expect:      []string{},
+			expectedErr: "",
+		},
+		"exclude_no_files": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, twoHoursAgo},
+			excludeOlderThan: 3 * time.Hour,
+
+			expect:      []string{"a.log", "b.log"},
+			expectedErr: "",
+		},
+		"exclude_some_files": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, threeHoursAgo},
+			excludeOlderThan: 3 * time.Hour,
+
+			expect:      []string{"a.log"},
+			expectedErr: "",
+		},
+		"exclude_all_files": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, threeHoursAgo},
+			excludeOlderThan: 90 * time.Minute,
+
+			expect:      []string{},
+			expectedErr: "",
+		},
+		"file_not_present": {
+			files:            []string{"a.log", "b.log"},
+			fileMTimes:       []time.Time{twoHoursAgo, {}},
+			excludeOlderThan: 3 * time.Hour,
+
+			expect:      []string{"a.log"},
+			expectedErr: "b.log: no such file or directory",
```

Comment:
Does this mean that a file was matched at first place and when we go to check its time with the additional filter it is not there any more?

Reply:
No, this test case is for `b.log` not existing at all, at any point in time. This test case ensures that the `ExcludeOlderThan` filter still works as expected in this scenario, by returning `a.log` since it's mtime was newer the `excludeOlderThan` duration but also returning an error that `b.log` was not found.

Reply:
Cool, my concern is mostly about the possibility to skip files at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31916/files#diff-8e42caa424da1bba5e24e65c825d5d69c87eda5652e5bc6d92f9e2ab6a9db897R190 in case they have been removed before this filter's apply. 
But in any case it seems that the matching process takes place all-together so not much that we can do here.

Reply:
The link doesn't bring up a specific line for me so I'm not sure what it was referring to. Anything we need to do about this?

Reply:
Ah sorry for this. My point was to link to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31916/files#diff-070f8096a08bc295ac03aaff7bb772c26241ff38f92a5d6f66ea7f6bf59b0113R22 where we actually skip a file if we don't find it.
But I think this would happen anyways when we reach at the reading phase and we will try to open the file, right?
So not really sth that can miss here.

Reply:
We can resolve that comment (for some reason I don't see the resolve button 🤔 might be a permission issue).